baseframe/forms/auto.py

```diff
@@ -54,18 +58,26 @@ def render_redirect(url, code=302):
         return redirect(url, code=code)
 
 
-def render_delete_sqla(obj, db, title, message, success=u'', next=None, cancel_url=None):
+def render_delete_sqla(obj, db, title, message, success=u'', next=None, cancel_url=None, with_response=True):
     if not obj:
         abort(404)
     form = ConfirmDeleteForm()
     if request.method in ('POST', 'DELETE') and form.validate():
+        print ('form', request.form)
         if 'delete' in request.form or request.method == 'DELETE':
             db.session.delete(obj)
             db.session.commit()
             if success:
                 flash(success, 'success')
+            if request.is_xhr:
+                return make_response(jsonify(status='ok', doc=_(u"Deleted"), result={'url': next or url_for('index')}), 303)
             return render_redirect(next or url_for('index'), code=303)
         else:
+            if request.is_xhr:
+                return make_response(jsonify(status='ok', doc=_(u"Cancel"), result={'url': cancel_url or next or url_for('index')}), 303)
             return render_redirect(cancel_url or next or url_for('index'), code=303)
-    template = THEME_FILES[current_app.config['theme']]['delete.html.jinja2']
-    return make_response(render_template(template, form=form, title=title, message=message))
+    if with_response:
+        template = THEME_FILES[current_app.config['theme']]['delete.html.jinja2']
+        return make_response(render_template(template, form=form, title=title, message=message))
+    template = THEME_FILES[current_app.config['theme']]['delete_template.html.jinja2']
+    return render_template(template, form=form, title=title, message=message)
```

Comment:
I don't get the difference between `delete.html.jinja2` and `delete_template.html.jinja2`. Could they have more descriptive names?

Reply:
Removing it for now. Could we look at `render_message` and `render_delete_sqla` as part crud-view PR?